Apps/Examples/Examples/All Examples/NavigationSimulatorExample.swift

```diff
@@ -0,0 +1,211 @@
+import UIKit
+import MapboxMaps
+
+final class NavigationSimulatorExample: UIViewController, ExampleProtocol {
+    private enum ID {
+        static let routeSource = "route-line-source-id"
+        static let routeLineLayer = "route-line-layer-id"
+        static let casingLineLayer = "route-casing-layer-id"
+    }
+
+    private var mapView: MapView!
+    private var navigationSimulator: NavigationSimulator!
+
+    private lazy var routeSource: Source = {
+        var source = GeoJSONSource()
+        source.data = .geometry(Geometry(sampleRouteLine))
+        source.lineMetrics = true
+
+        return source
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        mapView = MapView(frame: view.bounds)
+        configureMap()
+
+        view.addSubview(mapView)
+        mapView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            mapView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            mapView.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor),
+            mapView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+            mapView.rightAnchor.constraint(equalTo: view.safeAreaLayoutGuide.rightAnchor),
```

Comment:
interesting, we should actually constraint or bind the map to the safe area and not the whole view controller, since it is configured for its contents to be extended under the navigation bar -> we can't see the top of the map as Andrew pointed out here https://github.com/mapbox/mapbox-maps-ios/pull/1225#discussion_r838886458
I think the way navigation bar looks like that when view is constrained to navigation bar is because of the way we configure `UINavigationBarAppearance` for iOS 13+

this snippet fixed the problem though, wdyt?
```
        if #available(iOS 13.0, *) {
            let navigationBarAppearance = UINavigationBarAppearance()
            navigationBarAppearance.configureWithOpaqueBackground()

            appearance.scrollEdgeAppearance = navigationBarAppearance
            appearance.compactAppearance = navigationBarAppearance
            appearance.standardAppearance = navigationBarAppearance
        }
```

Reply:
I think extending the map under the bars and making it fill the whole screen makes it looks like it belongs on iOS(the map is visible under the translucent bars). I checked how apple handles this in their Maps app, they use the globe view on lower zoom levels that eliminates this issue. Google Maps seems to both use opaque bars **and** have a part of the map behind them(obstructing a portion of the map behind them, of course).

Reply:
for the sake of example I think it makes more sense to display the full map without any part that is hidden, since we don't really have anything to show/hide navigation bar so it looks weird if a part of the map is not visible.

| Full | Safe-area |
| --- | --- |
| ![Simulator Screen Shot - iPhone 11 - 2022-06-16 at 12 10 38](https://user-images.githubusercontent.com/30261043/174036377-a4f661cd-1881-4133-b6c2-a217a676138a.png) | ![Simulator Screen Shot - iPhone 11 - 2022-06-16 at 12 10 57](https://user-images.githubusercontent.com/30261043/174036422-2c922073-fa56-45bb-9d28-32bb8bb52c7c.png) |